pkg/scheduler/scheduler.go

```diff
@@ -324,23 +325,46 @@ type partialAssignment struct {
 	preemptionTargets []*workload.Info
 }
 
+// resourcesToReserve calculates how much of available resources in cq assignment could reserve.
+func resourcesToReserve(e *entry, cq *cache.ClusterQueue) cache.FlavorResourceQuantities {
+	if e.CanBePartiallyAdmitted() || e.assignment.Borrowing {
+		return e.assignment.Usage
+	}
+	reservedUsage := e.assignment.Usage
+	for flavor, resourceUsage := range e.assignment.Usage {
+		cqQuota := make(map[corev1.ResourceName]*cache.ResourceQuota)
+		for _, rg := range cq.ResourceGroups {
+			for _, cqFlavor := range rg.Flavors {
+				if cqFlavor.Name == flavor {
+					cqQuota = cqFlavor.Resources
+				}
+			}
+		}
+		for resource, usage := range resourceUsage {
+			// since assignment doesn't borrow, it's usage can't exceed nominal quota.
+			reservedUsage[flavor][resource] = min(usage, cqQuota[resource].Nominal-cq.Usage[flavor][resource])
```

Comment:
IIUC, this calculation always returns `usage` since the assignment doesn't allow partial admission, right?

Reply:
No, in case https://github.com/kubernetes-sigs/kueue/issues/1283#issuecomment-1834370396 it returns 0 for workload 3 instead of 1 (nominal quota - cq usage = 1-1). 
Here `usage` is accumulated resources requests.

Reply:
Oops, you're right. The entry's representativeMode isn't only `Fit` even if the cq doesn't belong to any cohort.
Thanks :)